services/management/dbaas/components_service.go

```diff
@@ -96,6 +106,71 @@ func (c componentsService) GetPXCComponents(ctx context.Context, req *dbaasv1bet
 	return &dbaasv1beta1.GetPXCComponentsResponse{Versions: versions}, nil
 }
 
+// shouldFilterOutVersion decides if given version should be filtered out of response with database versions.
+func shouldFilterOutVersion(disabledVersions map[string]disablingModifier, version string) (bool, error) {
+	parsedVersion, err := goversion.NewVersion(version)
+	if err != nil {
+		return false, err
+	}
+	for disabledVersion, modifier := range disabledVersions {
+		switch modifier {
+		case equal:
+			if len(disabledVersion) <= len(version) {
+				if version[:len(disabledVersion)] == disabledVersion {
+					return true, nil
+				}
+			}
+		case less:
+			parsedDisabledVersion, err := goversion.NewVersion(disabledVersion)
+			if err != nil {
+				return false, err
+			}
+			if parsedVersion.LessThan(parsedDisabledVersion) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// removeDisabledVersions removes disabled versions from the given varsions matrix.
+func (c componentsService) removeDisabledVersions(operator string, m matrix) error {
+	// TODO: load this map from database.
+	disabledVersions := map[string](map[string]disablingModifier){
+		PXCOperator: map[string]disablingModifier{
+			"5.7": equal,
+		},
+		PSMDBOperator: map[string]disablingModifier{
+			"4.2": less,
+		},
+	}
+	var versions map[string]component
+	switch operator {
+	case PSMDBOperator:
+		versions = m.Mongod
+	case PXCOperator:
+		versions = m.Pxc
+	default:
+		return errors.Errorf("failed to filter out disabled versions, unknown operator %s", operator)
+	}
+
+	versionsToDelete := make([]string, 0)
+	disabledVersionsPerOperator := disabledVersions[operator]
+	for version := range versions {
+		filterOut, err := shouldFilterOutVersion(disabledVersionsPerOperator, version)
+		if err != nil {
+			return err
+		}
+		if filterOut {
+			versionsToDelete = append(versionsToDelete, version)
```

Comment:
do we need this? you can remove it from map here

Reply:
removed